test/download_helper.js

```diff
@@ -0,0 +1,257 @@
+      }
+
+      dh.createButton();
+      
+      {
+        const button = document.querySelectorAll(".downloadall")
+        assert(button.length == 1);
+        const idAttribute = button[0].getAttribute("id");
+        expect(idAttribute == "someId").to.be.true;
+      }
+    });
+
+    it('should create specific properties for the button', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles"></div>
+      `)
+      
+      dh.createButton();
+
+      const buttonClassName = "downloadall"
+      expect(dh.button.className == buttonClassName).to.be.true;
+
+      const buttonTitleExpected = "Generates a file for automating downloads using 'curl'"
+      expect(dh.button.title == buttonTitleExpected).to.be.true;
+
+      expect(dh.button.disabled).to.be.true;
+
+      expect(dh.button.textContent == "preparing download").to.be.true;
+    });
+  });
+
+  describe("enableButton()", () => {
+    it('button should call functions on click', async () => {
+      DownloadHelper.dateString = sinon.fake.returns("dateString");
+      DownloadHelper.generateDownloadList = sinon.fake.returns("downloadList");
+      DownloadHelper.download = sinon.spy();
+
+      dh.button = document.createElement("button");
+
+      dh.enableButton();
+
+      dh.button.click();
+
+      expect(DownloadHelper.dateString.calledOnce).to.be.true;
+      expect(DownloadHelper.generateDownloadList.calledOnce).to.be.true;
+      
+      expect(DownloadHelper.download.calledOnce).to.be.true;
+      expect(DownloadHelper.download.getCall(0).args[0] == "bandcamp_dateString.txt").to.be.true;
+      expect(DownloadHelper.download.getCall(0).args[1] == preamble + "downloadList").to.be.true;
+    });
+
+    it('button should have specific properties', async () => {
+      dh.button = document.createElement("button");
+
+      dh.enableButton();
+
+      expect(dh.button.disabled).to.be.false;
+      expect(dh.button.textContent == "Download curl File").to.be.true;
+    });
+  });
+
+  describe("disableButton()", () => {
+    it('button listener should be removed', async () => {
+      dh.button = document.createElement("button")
+      dh.button.removeEventListener = sinon.spy();
+
+      dh.disableButton();
+
+      expect(dh.button.removeEventListener.calledOnce).to.be.true;
+
+    });
+    it('button should have specific properties', async () => {
+      dh.button = document.createElement("button")
+
+      dh.disableButton();
+
+      expect(dh.button.disabled).to.be.true;
+      expect(dh.button.textContent == "preparing download").to.be.true;
+    });
+  });
+
+  describe("callback()", () => {
+    it('enables button when all style.display blank', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-title"><div class="item-button" style="display: none"></div></div>
+        <div id="testId" class="download-title"><div class="item-button" style="display: none"></div></div>
```

Comment:
Technically invalid HTML here as `id` is supposed to be unique within the DOM. That might be an interesting test case...